src/toil_scripts/gatk_germline/germline.py

```diff
@@ -21,198 +22,114 @@
+from toil_scripts.tools.preprocessing import run_gatk_preprocessing, run_samtools_sort, \
     run_samtools_index, run_samtools_view, run_picard_create_sequence_dictionary
+from toil_scripts.tools.variant_annotation import run_oncotator
+from toil_scripts.tools.variant_filters import split_vcf_by_name
 
 
-def download_and_run(job, uuid, url, config, rg_line=None):
+def parse_manifest(path_to_manifest):
     """
-    Downloads reference files and runs the GATK best practices germline pipeline for a single
-    samples.
+    Parses samples, specified in either a manifest or listed with --samples
 
-    :param JobFunctionWrappingJob job: Toil Job instance
-    :param str uuid: Unique identifier for the sample
-    :param str url: URL to BAM file
-    :param Namespace config: Configuration options for pipeline
-    :param str rg_line: RG line for BWA alignment. Default is None.
-    :return: None
+    :param path_to_manifest str: Path to configuration file
+    :return: Samples and their attributes as defined in the manifest
+    :rtype: list[list]
     """
-    get_shared_files = job.wrapJobFn(download_shared_files, config).encapsulate()
-    samples = [(uuid, url, rg_line)]
-    run_pipeline = Job.wrapJobFn(run_gatk_germline_pipeline, samples, get_shared_files.rv())
-    get_shared_files.addChild(run_pipeline)
+    bam_re = r"^(?P<uuid>\S+)\s(?P<url>\S+[bsc][r]?am)"
+    fq_re = r"^(?P<uuid>\S+)\s(?P<url>\S+)\s(?P<url2>\S+)?\s?(?P<rg_line>@RG\S+)"
+    samples = []
+    with open(path_to_manifest, 'r') as f:
+        for line in f.readlines():
+            line = line.strip()
+            if line.startswith('#'):
+                continue
+            bam_match = re.match(bam_re, line)
+            fastq_match = re.match(fq_re, line)
+            if bam_match:
+                uuid = bam_match.group('uuid')
+                url = bam_match.group('url')
+                url2 = None
+                rg_line = None
+                require('.bam' in url.lower() or '.fastq' in url.lower(),
+                        'Expected .bam extension:\n{}:\t{}'.format(uuid, url))
+            elif fastq_match:
+                uuid = fastq_match.group('uuid')
+                url = fastq_match.group('url')
+                url2 = fastq_match.group('url2')
+                rg_line = fastq_match.group('rg_line')
+                require('.fq' in url.lower() or '.fastq' in url.lower(),
+                        'Expected .fq extension:\n{}:\t{}'.format(uuid, url))
+            else:
+                msg = 'Could not parse entry in manifest: %s\n%s' % (f.path, line)
+                raise ValueError(msg)
+            require(urlparse(url).scheme or url.startswith('syn'),
+                    'Invalid URL passed for {}'.format(url))
+            samples.append((uuid, url, url2, rg_line))
+    return samples
 
 
-def run_gatk_germline_pipeline(job, samples, config):
+def convert_space_resource(attribute, default):
```

Comment:
Ping @hannes-ucsc but I'm pretty sure this exists already as a library function.